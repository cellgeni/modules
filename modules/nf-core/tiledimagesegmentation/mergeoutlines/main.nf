process TILEDIMAGESEGMENTATION_MERGEOUTLINES {
    tag "$meta.id"

    container 'quay.io/cellgeni/imagetileprocessor:0.1.11'

    input:
    tuple val(meta), path(outlines)

    output:
    tuple val(meta), path("${prefix}.geojson"), emit: multipoly_geojsons
    tuple val(meta), path("${prefix}.wkt"), emit: multipoly_wkts, optional: true
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_merged"
    """
    merge-polygons \\
        --wkts $outlines \\
        --output_prefix "${prefix}" \\
        $args \\

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mergeoutlines: \$(merge-polygons --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_merged"
    """
    touch ${prefix}.geojson

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mergeoutlines: \$(merge-polygons --version)
    END_VERSIONS
    """
}
