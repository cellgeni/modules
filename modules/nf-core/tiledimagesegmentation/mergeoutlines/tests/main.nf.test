nextflow_process {

    name "Test Process TILEDIMAGESEGMENTATION_MERGEOUTLINES"
    script "../main.nf"
    process "TILEDIMAGESEGMENTATION_MERGEOUTLINES"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "tiledimagesegmentation"
    tag "tiledimagesegmentation/mergeoutlines"

    test("mouse heart - wkt") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    [
                        file(params.modules_testdata_base_path + 'test-0_0_200_200-diam_20_cp_outlines.wkt', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'test-0_0_500_500-diam_20_cp_outlines.wkt', checkIfExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("mouse heart - wkt - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    [
                        file(params.modules_testdata_base_path + 'test-0_0_200_200-diam_20_cp_outlines.wkt', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'test-0_0_500_500-diam_20_cp_outlines.wkt', checkIfExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
