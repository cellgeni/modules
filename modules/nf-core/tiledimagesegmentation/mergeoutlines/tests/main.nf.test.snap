{
    "mouse heart - wkt - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,15bd7624a79e500fc090c787cf0a8e53"
                ],
                "multipoly_geojsons": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multipoly_wkts": [
                    
                ],
                "versions": [
                    "versions.yml:md5,15bd7624a79e500fc090c787cf0a8e53"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T20:11:26.546806854"
    },
    "mouse heart - wkt": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,0e17590b6af5065d58270ee8708db50c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,ac7ba24ae1deff214bf465aed36a42f9"
                    ]
                ],
                "2": [
                    "versions.yml:md5,15bd7624a79e500fc090c787cf0a8e53"
                ],
                "multipoly_geojsons": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.geojson:md5,0e17590b6af5065d58270ee8708db50c"
                    ]
                ],
                "multipoly_wkts": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged.wkt:md5,ac7ba24ae1deff214bf465aed36a42f9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,15bd7624a79e500fc090c787cf0a8e53"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-28T20:11:15.676181083"
    }
}